common/src/test/java/com/github/dominik48n/party/server/SwitchServerTest.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2023 Dominik48N
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.github.dominik48n.party.server;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.github.dominik48n.party.api.Party;
+import com.github.dominik48n.party.api.PartyAPI;
+import com.github.dominik48n.party.api.PartyProvider;
+import com.github.dominik48n.party.api.player.PartyPlayer;
+import com.github.dominik48n.party.config.SwitchServerConfig;
+import com.github.dominik48n.party.user.UserManager;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.platform.commons.util.ReflectionUtils;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import java.lang.reflect.Field;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.regex.Pattern;
+import static java.util.List.of;
+import static org.junit.platform.commons.util.ReflectionUtils.HierarchyTraversalMode.TOP_DOWN;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.never;
+
+@ExtendWith(MockitoExtension.class)
+class SwitchServerTest {
+
+    private SwitchServer<String> switchServer;
+
+    @Mock
+    private UserManager<String> userManager;
+
+    @Mock
+    private SwitchServerConfig switchServerConfig;
+
+    @Mock
+    private PartyPlayer partyPlayer;
+
+    @Mock
+    private PartyProvider partyProvider;
+
+    @BeforeEach
+    void setUp() throws IllegalAccessException {
+        setPartyProvider();
+        switchServer = createSwitchServer();
+    }
+
+    @Test
+    void handleServerConnected() throws JsonProcessingException {
```

Comment:
Please declare the variables in the method that should not be changed as `final`.